pkg/controllermanager/controller/project/project.go

```diff
@@ -97,9 +101,14 @@ func NewProjectController(
 		DeleteFunc: func(obj interface{}) { projectController.roleBindingDelete(ctx, obj) },
 	})
 
+	shootInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) { projectController.updateLastActivityTimestamp(ctx, obj) },
```

Comment:
The `AddFunc` will be called every time the GCM is started, i.e. it will iterate over all shoots and updates the `LastActivityTimestamp` which is probably not what we want. Instead, I suggest to register and `UpdateFunc` and renew the `LastActivityTimestamp` when the shoot was first scheduled on a seed.

Reply:
Make sense, but I am not sure that this will work if the user creates a shoot with `spec.SeedName:<seed>`. 
Also, the update function will be called far more often and I don't think that we need to know about every update of the shoot, or other resources, but only when new resources are created. 
I will try a solution where the `AddFunc`  will compare the `CreationTimestamp` of the object with the `LastActivityTimestamp`, and then it will update the last activity only if the creation timestamp is newer, if that make sense to you.

Reply:
Okay thanks 👍  Maybe it's also worth a thought to write a separate reconciler for this concern because it enables you to react upon error, e.g. you can re-queue the request in case an update error happened. WDYT?

Reply:
I have played around with the code to see what will turn out and I don't think that we should overcomplicate by adding another reconciler just for this field in the project status. I would prefer if we stick to that approach that is similar to what is done with the [RoleBInding update](https://github.com/gardener/gardener/blob/master/pkg/controllermanager/controller/project/rolebinding_control.go)

Reply:
> I would prefer if we stick to that approach that is similar to what is done with the RoleBInding update

The difference is that "RoleBInding update" only does a mapping while your event handler also mutates a resource. There won't be any retries if that update fails, i.e. a new timestamp will only be set when another shoot is created or GCM is restarted.

I think we ideally had:
1) Reconciler which updates the `status.lastActivityTimestamp` whenever a relevant resource is created (shoot)
2) Update from (1) triggers the `ProjectStaleReconciler` which marks the project as not stale

This has the following benefits even though with a bigger boilerplate code:
- Retry in case of an error of lastActivityTimestamp
- Tests for new reconciler can be added easily (whereas there are no tests in this PR for the event handler)
- Isolation & separation of concern
- Follows controller best practices